homeassistant/components/godice/sensor.py

```diff
@@ -0,0 +1,136 @@
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Setups Dice sensors."""
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    device = data[DATA_DEVICE]
+    device_info = data[DATA_DEVICE_INFO]
+
+    entry_ctors = [
+        DiceColorSensor,
+        BatteryLevelSensor,
+        DiceNumberSensor,
+    ]
+
+    entries = [ctor(device_info, device) for ctor in entry_ctors]
+    async_add_entities(entries)
+
+
+class BaseSensor(SensorEntity):
+    """Base for concrete sensors."""
+
+    _attr_should_poll = False
+    _attr_has_entity_name = True
+
+    def __init__(
+        self, entity_description: SensorEntityDescription, device_info: DeviceInfo
+    ) -> None:
+        """Set default values."""
+        self.entity_description = entity_description
+        self._attr_device_info = device_info
+        self._attr_unique_id = (
+            f"{device_info[const.CONF_NAME]}_{self.entity_description.key}"
+        )
+        self._attr_translation_key = entity_description.key
+
+
+class DiceColorSensor(RestoreSensor, BaseSensor):
+    """Represents color of a dice (dots)."""
+
+    dice: godice.Dice
+
+    def __init__(self, device_info: DeviceInfo, device: godice.Dice) -> None:
+        """Set default values."""
+        super().__init__(COLOR_SENSOR_DESCR, device_info)
+        self.dice = device
+
+    async def async_added_to_hass(self) -> None:
+        """Restore a color value if any, read the value otherwise."""
+        last_state = await self.async_get_last_sensor_data()
+        if last_state and last_state.native_value is not None:
+            self._attr_native_value = last_state.native_value
+        else:
+            color = await self.dice.get_color()
+            self._attr_native_value = color.name
+
+
+class BatteryLevelSensor(BaseSensor):
+    """Represents battery level."""
+
+    _attr_should_poll = True
+    dice: godice.Dice
+
+    def __init__(self, device_info: DeviceInfo, device: godice.Dice) -> None:
+        """Set default values."""
+        super().__init__(BATTERY_SENSOR_DESCR, device_info)
+        self.dice = device
+
+    async def async_update(self) -> None:
+        """Poll battery level."""
+        _LOGGER.debug("Battery level update")
+        self._attr_native_value = await self.dice.get_battery_level()
+
+
+class DiceNumberSensor(BaseSensor):
+    """Represents the rolled dice number."""
+
+    dice: godice.Dice
+
+    def __init__(self, device_info: DeviceInfo, device: godice.Dice) -> None:
+        """Set default values."""
+        super().__init__(ROLLED_NUMBER_SENSOR_DESCR, device_info)
+        self.dice = device
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe on rolled number events."""
+        await self.dice.subscribe_number_notification(self._handle_upd)
+
+    async def _handle_upd(
+        self, number: int, _stability_descr: godice.StabilityDescriptor
+    ) -> None:
```

Comment:
I would suggest just writing names out fully when they are not too long. Makes it much easier to understand what the function does when quickly reading through it. Same goes for a couple of other cases like using 'desc' for 'description', or 'devinfo' for 'device_info' which you already addressed. Difference in length is not too much but something like device_info or _handle_number_update is much clearer as opposed to devinfo or _handle_upd
```suggestion
    async def _handle_number_update(
        self, number: int, _stability_descr: godice.StabilityDescriptor
    ) -> None:
```